### PR TITLE
Feature/366/create recipe ingredients

### PR DESCRIPTION
**Pull request by andreaMT15**
### Ticket Number:
366

### Describe The Problem Being Solved:
User can now add ingredients to a recipe

### Checklist For Submitter

* [x ] I have documented all new functionality
* [x] I have screenshotted new UI changes and attached them to this PR
Screen Shot 2019-10-24 at 8 28 23 PM

### CODE REVIEW

File: app/controllers/api/v1/recipes_controller.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def create
   end
 
   def recipe_params
-   params.require(:recipe).permit(:title, :snippet, :difficulty, :duration, :servings )
+   params.require(:recipe).permit(:title, :snippet, :difficulty, :duration, :servings, ingredients: [])
   end
 
 end
```
Comment (H12) [inline app/controllers/api/v1/recipes_controller.rb]: As long as we're here... Let's add a newline to the end of this file.

File: frontend/src/components/EditRecipe/EditRecipeForm.css [hunk 2]
```diff
@@ -48,6 +48,20 @@
...
 
+.ingredient-btn-container{
+  display: flex;
+  justify-content: flex-end;
+  width: 70%;
+}
+
+.add-ingredient-button {
+  display: inline-block;
+  margin: 20px 20px;
```
Comment (H12) [inline frontend/src/components/EditRecipe/EditRecipeForm.css]: You can just do `margin: 20px;` here, right?

File: frontend/src/components/EditRecipe/EditRecipeForm.css [hunk 3]
```diff
@@ -48,6 +48,20 @@
...
+  border: 2px solid #ffa616;
+  margin-top: 20px;
```
Comment (H12) [inline frontend/src/components/EditRecipe/EditRecipeForm.css]: And if you do the above, I think you can get rid of this too.

File: frontend/src/containers/RecipePage/Recipe.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ class Recipe extends Component {
       recipeId, directions, title, ingredients, snippet, difficulty, duration, servings, photo,
       userId, comments,
     } = this.state;
-
+    console.log(ingredients);
```
Comment (H12) [inline frontend/src/containers/RecipePage/Recipe.js]: 🗑

File: frontend/src/components/EditRecipe/EditRecipeForm.js [hunk 5]
```diff
@@ -76,6 +99,32 @@ const EditRecipeForm = ({ history }) => {
           </select>
         </div>
       </div>
+      <div>
+        <div className="ingredients-container">
+          <h1>Ingredients</h1>
+
```
Comment (H12) [inline frontend/src/components/EditRecipe/EditRecipeForm.js]: I think this newline is unnecessary.

File: frontend/src/components/EditRecipe/EditRecipeForm.js [hunk 6]
```diff
@@ -11,16 +12,38 @@ const EditRecipeForm = ({ history }) => {
...
+    const values = [...ingredients];
+    values[i] = event.target.value;
+    setIngredients(values);
+  };
+  const handleAdd = () => {
+    const values = [...ingredients];
+    values.push('');
+    setIngredients(values);
+  };
+  const handleRemove = (i) => {
```
Comment (vhoof) [inline frontend/src/components/EditRecipe/EditRecipeForm.js]: I'd probably change the names of all these functions to be `handleIngredientChange`, etc. We'll probably need other `handleChange` functions on this page.

Comment (nickharriscodes) [reply frontend/src/components/EditRecipe/EditRecipeForm.js]: What would you think about moving these functions and corresponding lines of JSX into their own separate file? I was getting started on the recipe description ticket and am realizing how unwieldy this EditRecipeForm is becoming.

Comment (nickharriscodes) [reply frontend/src/components/EditRecipe/EditRecipeForm.js]: @vhoof

File: frontend/src/components/EditRecipe/EditRecipeForm.js [hunk 7]
```diff
@@ -76,11 +100,19 @@ const EditRecipeForm = ({ history }) => {
...
-        <button type="submit" id="recipe-submit-btn" value="submit" onClick={handleSubmit}>Submit</button>
-      </div>
-      <div className="row form-footer">
-        <Footer />
+      <div>
+        <Ingredients
+          ingredients={ingredients}
+          handleIngredientChange={handleIngredientChange}
+          handleIngredientAdd={handleIngredientAdd}
+          handleIngredientRemove={handleIngredientRemove}
```
Comment (vhoof) [inline frontend/src/components/EditRecipe/EditRecipeForm.js]: What do you think about passing in `setIngredients` and doing these `handleChange`, etc. functions in the ingredients component?

**Comment by andreaMT15** [comment]
@H12 Can you please re-review?

File: frontend/src/components/EditRecipe/EditRecipeForm.js [hunk 8]
```diff
@@ -109,7 +112,7 @@ const EditRecipeForm = ({ history }) => {
               onChange={(selected) => {
                 setSelectedTags(selected);
               }
-            }
+              }
```
Comment (H12) [inline frontend/src/components/EditRecipe/EditRecipeForm.js]: I think you want this on the previous line.

File: frontend/src/components/EditRecipe/Ingredients/Ingredients.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    const values = [...ingredients];
+    values[i] = event.target.value;
+    setIngredients(values);
+  };
+
+  const handleIngredientAdd = () => {
+    const values = [...ingredients];
+    values.push('');
+    setIngredients(values);
+  };
```
Comment (H12) [inline frontend/src/components/EditRecipe/Ingredients/Ingredients.js]: Let's add a newline between these two functions.

File: frontend/src/components/EditRecipe/Ingredients/Ingredients.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import map from 'lodash/map';
+import './ingredients.css';
+
+
```
Comment (H12) [inline frontend/src/components/EditRecipe/Ingredients/Ingredients.js]: This newline is unnecessary.

**Pull request merged**